cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go

```diff
@@ -184,23 +185,28 @@ func runControlPlanePrepareControlPlaneSubphase(c workflow.RunData) error {
 		return err
 	}
 
-	fmt.Printf("[control-plane] Using manifest folder %q\n", kubeadmconstants.GetStaticPodDirectory())
+	fmt.Printf("[control-plane] Using manifest folder %q\n", data.ManifestDir())
+
+	// If we're dry-running, set CertificatesDir to default value to get the right cert path in static pod yaml
+	if data.DryRun() {
+		cfg.CertificatesDir = filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.DefaultCertificateDir)
+	}
+
 	for _, component := range kubeadmconstants.ControlPlaneComponents {
 		fmt.Printf("[control-plane] Creating static Pod manifest for %q\n", component)
 		err := controlplane.CreateStaticPodFiles(
-			kubeadmconstants.GetStaticPodDirectory(),
+			data.ManifestDir(),
 			data.PatchesDir(),
 			&cfg.ClusterConfiguration,
 			&cfg.LocalAPIEndpoint,
-			// TODO: add support for join dry-run:
-			// https://github.com/kubernetes/kubeadm/issues/2505
-			false,
+			data.DryRun(),
 			component,
 		)
 		if err != nil {
 			return err
 		}
 	}
+
```

Comment:
```suggestion
```

Reply:
I will check the redundant null line too, fixed